extensions/ql-vscode/src/remote-queries/run-remote-query.ts

```diff
@@ -225,7 +224,17 @@ export async function prepareRemoteQueryRun(
     throw new UserCancellationException('Cancelled');
   }
 
-  const { base64Pack, language } = await generateQueryPack(cliServer, queryFile, queryPackDir);
+  const { remoteQueryDir, queryPackDir } = await createRemoteQueriesTempDirectory();
+
+  let pack: Awaited<ReturnType<typeof generateQueryPack>>;
```

Comment:
This is fine but I personally generally avoid using this sort of type and go with defining a type for the return type of the function. E.g. have a `QueryPack` interface and use that in the `generateQueryPack` signature as well as the definition of the `pack` variable.

What do you think of updating `generateQueryPack` to return a pre-defined t